service-commands/src/setup.js

```diff
@@ -554,83 +588,101 @@ const identityServiceUp = async (options = { verbose: false }) => {
-    [Service.SOLANA_VALIDATOR, SetupCommand.UP]
+    [Service.SOLANA_VALIDATOR, SetupCommand.UP],
+    [Service.SOLANA_VALIDATOR, SetupCommand.HEALTH_CHECK]
   ]
 
-  const inParallel = [
+  const ipfsAndContractsCommands = [
     [Service.IPFS, SetupCommand.UP],
     [Service.IPFS_2, SetupCommand.UP],
     [Service.CONTRACTS, SetupCommand.UP],
     [Service.ETH_CONTRACTS, SetupCommand.UP],
-    [Service.SOLANA_PROGRAMS, SetupCommand.UP]
+    [Service.SOLANA_PROGRAMS, SetupCommand.UP],
   ]
 
-  let creatorNodeCommands = _.range(1, numCreatorNodes + 1).map(serviceNumber => {
-    return [
-      [
-        Service.CREATOR_NODE,
-        SetupCommand.UPDATE_DELEGATE_WALLET,
-        { serviceNumber, ...options }
-      ],
-      [
-        Service.CREATOR_NODE,
-        SetupCommand.UP,
-        { serviceNumber, ...options, waitSec: 10 }
-      ],
-      [
-        Service.CREATOR_NODE,
-        SetupCommand.HEALTH_CHECK,
-        { serviceNumber, ...options }
-      ],
-      [
-        Service.CREATOR_NODE,
-        SetupCommand.REGISTER,
-        { serviceNumber, ...options }
+  let creatorNodesCommands = _.range(1, numCreatorNodes + 1).map(
+    serviceNumber => {
+      return [
+        [
+          Service.CREATOR_NODE,
+          SetupCommand.UPDATE_DELEGATE_WALLET,
+          { serviceNumber, ...options }
+        ],
+        [
+          Service.CREATOR_NODE,
+          SetupCommand.UP,
+          { serviceNumber, ...options, waitSec: 10 }
+        ],
+        [
+          Service.CREATOR_NODE,
+          SetupCommand.HEALTH_CHECK,
+          { serviceNumber, ...options }
+        ],
+        [
+          Service.CREATOR_NODE,
+          SetupCommand.REGISTER,
+          { serviceNumber, ...options }
+        ]
       ]
-    ]
-  })
+    }
+  )
 
-  let discoveryNodesCommands = _.range(1, numDiscoveryNodes + 1).map(serviceNumber => {
-    return [
-      [
-        Service.DISCOVERY_PROVIDER,
-        SetupCommand.UP,
-        { serviceNumber, ...options }
-      ],
-      [
-        Service.DISCOVERY_PROVIDER,
-        SetupCommand.HEALTH_CHECK,
-        { serviceNumber, ...options }
-      ],
-      [
-        Service.DISCOVERY_PROVIDER,
-        SetupCommand.REGISTER,
-        { retries: 2, serviceNumber, ...options }
+  let discoveryNodesCommands = _.range(1, numDiscoveryNodes + 1).map(
+    serviceNumber => {
+      return [
+        [
+          Service.DISCOVERY_PROVIDER,
+          SetupCommand.UP,
+          { serviceNumber, ...options }
+        ],
+        [
+          Service.DISCOVERY_PROVIDER,
+          SetupCommand.HEALTH_CHECK,
+          { serviceNumber, ...options }
+        ],
+        [
+          Service.DISCOVERY_PROVIDER,
+          SetupCommand.REGISTER,
+          { retries: 2, serviceNumber, ...options }
+        ]
       ]
-    ]
-  })
+    }
+  )
 
   const sequential1 = [
     [Service.INIT_CONTRACTS_INFO, SetupCommand.UP],
+    [Service.USER_REPLICA_SET_MANAGER, SetupCommand.UP],
```

Comment:
moved URSM init earlier, so now only there should be only one instance of `URSM init failed bc endpoint not found` in logs - the one that comes from URSM init itself. i'll look into this later to find if there's a way to suppress this chicken/egg issue from logging an error when URSM fails to find itself when init'ing libs

Reply:
i think the libs ctor should probably allow some sort of URSM config that just tells it that it doesn't need a URSM at all to initialize. we do that for other services. e.g. libs can init fully without even a content node config or identity config https://github.com/AudiusProject/audius-protocol/blob/master/libs/src/index.js#L308